src/evict.c

```diff
@@ -555,13 +556,18 @@ int performEvictions(void) {
                 /* We don't want to make local-db choices when expiring keys,
                  * so to start populate the eviction pool sampling keys from
                  * every DB. */
-                for (i = 0; i < server.dbnum; i++) {
-                    db = server.db+i;
+                int samples = 0;
+                int i = (uint32_t)random()%server.dbnum;
+                int count = i+server.dbnum;
+                for (; i < count; i++) {
+                    int dbid = i%server.dbnum;
+                    db = server.db + dbid;
                     dict = (server.maxmemory_policy & MAXMEMORY_FLAG_ALLKEYS) ?
                             db->dict : db->expires;
                     if ((keys = dictSize(dict)) != 0) {
-                        evictionPoolPopulate(i, dict, db->dict, pool);
+                        samples += evictionPoolPopulate(dbid, dict, db->dict, pool, (server.maxmemory_samples+server.dbnum-1)/server.dbnum);
```

Comment:
Why not just shrink server.maxmemory_samples by server.dbnum times?

Reply:
Big server.maxmemory_samples  will lead to slowlog.